File: solutions/1656/java/Solution.java
// 1656. Design an Ordered Stream
import java.util.List;
import java.util.ArrayList;

class OrderedStream
{
	private ArrayList<String> stream;
	private int currentChunkIndex = 0;

	public OrderedStream(int n)
	{
		stream = new ArrayList<String>();

		for (int i = 0; i < n; ++i)
		{
			stream.add("");
		}
	}

	public List<String> insert(int idKey, String value)
	{
		stream.set(idKey - 1, value);

		List<String> chunk = new ArrayList<String>();

		for (; currentChunkIndex < stream.size(); ++currentChunkIndex)
		{
			if (stream.get(currentChunkIndex) == "")
			{
				break;
			}

			chunk.add(stream.get(currentChunkIndex));
		}

		return chunk;
	}
}

File: solutions/1656/c++/solution.cpp
// 1656. Design an Ordered Stream
#include <vector>
#include <string>

using namespace std;

class OrderedStream
{
	vector<string> stream;
	int currentChunkIndex = 0;

	public:
	OrderedStream(int n) : stream(n, "") {}

	vector<string> insert(int idKey, string value)
	{
		stream[idKey - 1] = value;

		vector<string> chunk;

		for (; currentChunkIndex < stream.size(); ++currentChunkIndex)
		{
			if (stream[currentChunkIndex] == "")
			{
				break;
			}

			chunk.push_back(stream[currentChunkIndex]);
		}

		return chunk;
	}

	private:
};